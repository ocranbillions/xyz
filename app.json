{
  "addons": [
    "heroku-postgresql"
  ],
  "buildpacks": [
  ],
  "env": {
    "JWT_SECRET_KEY": {
      "required": true
    },
    "NPM_CONFIG_PRODUCTION": {
      "required": true
    }
  },
  "formation": {
  },
  "scripts": {
    "postdeploy": "npx sequelize-cli db:seed:all"
  },
  "name": "xyz",
  "stack": "heroku-18"
}
