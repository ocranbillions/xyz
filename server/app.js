import express from 'express';
import bodyParser from 'body-parser';
import { config } from 'dotenv';
import userRoutes from './routes/userRoutes';

config();

const app = express();

const port = process.env.PORT || 3000;

app.use(bodyParser.json());
app.use(bodyParser.urlencoded({ extended: true }));

// app.get('/', (req, res) => res.status(301).redirect('/api/v1'));

app.get('/', (req, res) => {
  res.send('Welcome to xzy');
});

// Handles user routes
app.use('/api/v1/users', userRoutes);


// app.use(ErrorHandler.sendError);


if (process.env.NODE_ENV !== 'test') {
  app.listen(port, () => console.log('Welcome to xyz'));
}
export default app;
