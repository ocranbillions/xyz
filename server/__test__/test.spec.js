/* eslint-disable no-undef */
import supertest from 'supertest';
import app from '../app';

const server = () => supertest(app);
const url = '/api/v1';

describe('Welcome to helo books', () => {
  it('Should return a message on app start', (done) => {
    server().get('/')
      .end((err, res) => {
        expect(res.statusCode).toEqual(200);
        expect(res.body.message).toEqual('Welcome to xyz');
        done();
        expect(res.body).toMatchSnapshot();
      });
  });
});
