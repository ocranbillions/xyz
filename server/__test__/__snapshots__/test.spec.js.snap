// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome to helo books Should return a message on app start 1`] = `
Object {
  "message": "Welcome to xyz",
}
`;
